### PR TITLE
Initial spike on discussions stub + controller

### PR DESCRIPTION
**Pull request by practicingruby**
Working on an initial spike of the discussions listing functionality.

### CODE REVIEW

File: app/controllers/discussions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    when "review"
+      Discussion.reviews
+    when "evaluation"
+      Discussion.evaluations
+    else
+      Discussion.conversations
+    end
+  end
+
+  def find_course
```
Comment (jordanbyron) [inline app/controllers/discussions_controller.rb:19]: This should probably be a private method. It helps identify / protect methods which are publicly available.

File: app/views/discussions/index.html.haml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+= link_to "Conversations", course_discussions_path(:category => "conversation")
+|
+= link_to "Reviews", course_discussions_path(:category => "review")
+|
+= link_to "Evaluations", course_discussions_path(:category => "evaluation")
```
Comment (jordanbyron) [inline app/views/discussions/index.html.haml:5]: This works without `@course` as an argument!? Rails magic :sparkles:

File: app/controllers/discussions_controller.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    @discussions = case params[:category]
+    when "conversation"
+      Discussion.conversations
+    when "review"
+      Discussion.reviews
+    when "evaluation"
+      Discussion.evaluations
+    else
+      Discussion.conversations
+    end
```
Comment (jordanbyron) [inline app/controllers/discussions_controller.rb:14]: What do you think about moving this logic down into the model? Something like 

``` ruby
class Discussion

  def self.for_category(category)
    case category
    when "conversation"
      Discussion.conversations
    when "review"
      Discussion.reviews
    when "evaluation"
      Discussion.evaluations
    else
      Discussion.conversations
    end
  end
end
```

Then in the controller we can do:

``` ruby
class DiscussionsController < ApplicationController
  def index
    @discussions = @course.discussions.for_category(params[:category])
  end
end
```

**Comment by jordanbyron** [comment]
Great start @sandal! Let me know what you think about my comments and if you need help moving on from here :bowtie:

File: app/controllers/discussions_controller.rb (hunk 1 above)
Comment (sindhri) [inline app/controllers/discussions_controller.rb:19]: just commited a fix for this

File: app/controllers/discussions_controller.rb (hunk 3 above)
Comment (practicingruby) [inline app/controllers/discussions_controller.rb:14]: I will probably do that eventually but for right now I'm not 100% sure that this is going to be our interface anyway. We will probably actually end up with `Discussion.search` which works like this:

``` ruby
@course.discussions.search(:archived => true, :category => "review")
```

I want to hold off refactoring until I know what the right logic will be for this.

**Comment by practicingruby** [comment]
@jordanbyron, we pushed some fixes based on your feedback. I think we should try merging this if you're OK with it, so that we can a) make sure that we keep integrating frequently so that merge conflicts are less likely and b) so that we have some functionality in master.

The code is very self-contained, so it shouldn't hurt to have it there. What do you think?

**Comment by jordanbyron** [comment]
@sandal / @sindhri those fixes look good to me. Merge away!

**Pull request merged**